### PR TITLE
LOG-1161: Added events for valid/invalid pipelines

### PR DESCRIPTION
**Pull request by igor-karpukhin**
### Description
CLO will emit events for a valid and invalid CLF pipelines

/cc @jcantrill 
/assign @jcantrill 

### Links
- JIRA: https://issues.redhat.com/browse/LOG-1161

### CODE REVIEW

**Comment by vimalk78** [comment]
Please add an example of added Event in the PR description

**Comment by igor-karpukhin** [comment]
@vimalk78 I'll a test with the example event output

**Comment by igor-karpukhin** [comment]
Here is an example event: 
````
Events:
  Type    Reason  Age                From                 Message
  ----    ------  ----               ----                 -------
  Normal  Ready   21s (x2 over 21s)  clusterlogforwarder  All pipelines are valid
```

File: pkg/controller/forwarding/forwarding_controller.go [hunk 1]
```diff
@@ -111,10 +117,13 @@ func (r *ReconcileForwarder) Reconcile(request reconcile.Request) (reconcile.Res
 
 	if instance.Status.IsReady() {
 		instance.Status.Conditions.SetCondition(condReady)
+		r.recorder.Event(instance, "Normal", string(condReady.Type), "All pipelines are valid")
```
Comment (alanconway) [inline pkg/controller/forwarding/forwarding_controller.go:120]: We don't want to emit an event for every reconcile, only when a condition changes:

```suggestion
		if instance.Status.Conditions.SetCondition(condReady) {
				r.recorder.Event(instance, "Normal", string(condReady.Type), "All pipelines are valid")
		}
```

File: pkg/controller/forwarding/forwarding_controller.go [hunk 2]
```diff
@@ -111,10 +117,13 @@ func (r *ReconcileForwarder) Reconcile(request reconcile.Request) (reconcile.Res
...
 	}
 
 	if instance.Status.IsDegraded() {
-		instance.Status.Conditions.SetCondition(condDegraded(logging.ReasonInvalid, "Some pipelines are degraded or invalid"))
+		msg := "Some pipelines are degraded or invalid"
+		instance.Status.Conditions.SetCondition(condDegraded(logging.ReasonInvalid, msg))
+		r.recorder.Event(instance, "Error", string(logging.ReasonInvalid), msg)
```
Comment (alanconway) [inline pkg/controller/forwarding/forwarding_controller.go:126]: Same again, only event if conition has changed.

**Review by alanconway — CHANGES_REQUESTED** [review]
We only want an event if the condition of the forwarder changes, not on every reconcile (see suggested diff) If we do that, then I think it's ok to have an event for "Ready" since it will only be sent on first start-up, and after recover from some degraded or invalid state.

File: pkg/controller/forwarding/forwarding_controller.go (hunk 1 above)
Comment (igor-karpukhin) [reply pkg/controller/forwarding/forwarding_controller.go]: Fixed

File: pkg/controller/forwarding/forwarding_controller.go (hunk 2 above)
Comment (igor-karpukhin) [reply pkg/controller/forwarding/forwarding_controller.go]: Fixed

**Review by alanconway — APPROVED** [review]
/lgtm

**Pull request merged**